src/Libraries/Microsoft.Extensions.AI.Abstractions/ChatCompletion/StreamingChatCompletionUpdateExtensions.cs

```diff
@@ -178,20 +193,6 @@ static void AddMessage(ChatCompletion completion, bool coalesceContent, KeyValue
             }
 
             completion.Choices.Add(entry.Value);
-
-            if (completion.Usage is null)
-            {
-                foreach (var content in entry.Value.Contents)
-                {
-                    if (content is UsageContent c)
-                    {
-                        completion.Usage = c.Details;
-                        entry.Value.Contents = entry.Value.Contents.ToList();
-                        _ = entry.Value.Contents.Remove(c);
-                        break;
-                    }
-                }
-            }
```

Comment:
This is only a semi-related bug fix. Now that we're tracking all token counts, we should be doing so across all messages when creating a ChatCompletion and not just for a single usage from a single choice.